backend/Wheelingful.Services/Outer/BookAuthorService.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Wheelingful.Core.Contracts.Auth;
+using Wheelingful.Core.Contracts.Books;
+using Wheelingful.Core.DTO.Books;
+using Wheelingful.Services.Extensions;
+
+namespace Wheelingful.Services.Outer;
+
+internal class BookAuthorService : IBookAuthorService
+{
+    private readonly IBookCoverManager _bookCover;
+    private readonly IBookManager _bookManager;
+    private readonly ICurrentUser _currentUser;
+
+    public BookAuthorService(IBookManager bookManager, IBookCoverManager bookCover, ICurrentUser currentUser)
+    {
+        _bookCover = bookCover;
+        _bookManager = bookManager;
+        _currentUser = currentUser;
+    }
+
+    public async Task CreateBook(NewBookModel model)
+    {
+        model.AuthorId = _currentUser.Id;
+
+        model.CoverId = await _bookCover.UploadCover(model.CoverBase64, model.Title, model.AuthorId);
+
+        await _bookManager.CreateBook(model);
+    }
+
+    public async Task UpdateBook(UpdatedBookModel model)
+    {
+        var oldCoverTask = _bookManager
```

Comment:
do you need to create a Task, if check is false

Reply:
Probably not. I was going to move on your request or leave it be otherwise.